### PR TITLE
BAEL-6848: Set a Parameter in a HttpServletRequest in Java

### PR DESCRIPTION
**Pull request by manfred106**
Fix tests in BAEL-6848

### CODE REVIEW

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 @RunWith(MockitoJUnitRunner.class)
 public class SanitizeParametersRequestWrapperUnitTest {
 
-    private String NEW_VALUE = "NEW VALUE";
+    private final String NEW_VALUE = "NEW VALUE";
```
Comment (theangrydev) [inline web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java:21]: private static final

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 @RunWith(MockitoJUnitRunner.class)
 public class SetParameterRequestWrapperUnitTest {
 
-    private String NEW_VALUE = "NEW VALUE";
+    private final String NEW_VALUE = "NEW VALUE";
```
Comment (theangrydev) [inline web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java:21]: private static final

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public void whenSetValueToWrapperParametersStringArray_thenThe2ndCallShouldNotEq
 
         firstCallValues[0] = NEW_VALUE;
         String[] secondCallValues = wrapper.getParameterValues("input");
-        assertNotEquals(firstCallValues, secondCallValues);
+        assertThat(firstCallValues).isNotEqualTo(secondCallValues);
```
Comment (theangrydev) [inline web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java:57]: I believe @KevinGilmore might have been trying to suggest you to do something like this:
```suggestion
        assertThat(secondCallValues).doesNotContain(NEW_VALUE);
```

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public void whenSetValueToWrapperParametersStringArray_thenThe2ndCallShouldNotEq
 
         firstCallValues[0] = NEW_VALUE;
         String[] secondCallValues = wrapper.getParameterValues("input");
-        assertNotEquals(firstCallValues, secondCallValues);
+        assertThat(firstCallValues).isNotEqualTo(secondCallValues);
```
Comment (theangrydev) [inline web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java:58]: I believe @KevinGilmore might have been trying to suggest you to do something like this:
```suggestion
        assertThat(secondCallValues).doesNotContain(NEW_VALUE);
```

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java (hunk 1 above)
Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java]: done

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java (hunk 2 above)
Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java]: done

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java (hunk 3 above)
Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java]: isNotEqualTo is correct.
We are asserting whether the 2nd call of getParaemterValues() is not equal to the modified value from the 1st call.

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java (hunk 4 above)
Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java]: isNotEqualTo is correct.
We are asserting whether the 2nd call of getParaemterValues() is not equal to the modified value from the 1st call.

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java (hunk 3 above)
Comment (theangrydev) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java]: I understand that isNotEqualTo correctly asserts that the 2nd call is not equal to the modified value from the 1st call

However, this is not really using any of the power of AssertJ to make the intention of the assertion clearer, which is what I think Kevin was trying to get you to take a look at

Logically, you are modifying with NEW_VALUE and then you want to assert that this change was not effective

A buggy implementation might for example append NEW_VALUE to the end of the array, and your isNotEqualTo assertion would be green even though there is a bug

By explicitly checking for NEW_VALUE being absent, you get a bit more test coverage, and the intention of the assertion is clearer

Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SanitizeParametersRequestWrapperUnitTest.java]: Assertion updated

File: web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java (hunk 4 above)
Comment (manfred106) [reply web-modules/javax-servlets-2/src/test/java/com/baeldung/setparam/SetParameterRequestWrapperUnitTest.java]: Assertion updated

**Review by theangrydev — APPROVED** [review]

**Pull request merged**